### PR TITLE
[Web API] Catch addListener errors

### PR DESCRIPTION
**Pull request by ViktoriiaKovalova**
Catch errors for chrome.smartCardProviderPrivate.addListener. Those can throw on the older versions of dev channel, i.e., when chrome.smartCardProviderPrivate didn't have all the events it currently has.

### CODE REVIEW

File: smart_card_connector_app/src/chrome-api-provider.js [hunk 1]
```diff
@@ -408,12 +408,21 @@ class ChromeEventListener extends goog.Disposable {
...
   }
 
   /** @override */
   disposeInternal() {
-    this.chromeEvent_.removeListener(this.callback_);
+    // If subscribed to chrome event, unsubscribe.
+    if (this.callback_) {
+      this.chromeEvent_.removeListener(this.callback_);
+    }
     this.callback_ = null;
```
Comment (emaxx-google) [inline smart_card_connector_app/src/chrome-api-provider.js:426]: nit: This line can also be moved inside `if`.

File: smart_card_connector_app/src/chrome-api-provider.js [hunk 2]
```diff
@@ -408,12 +408,21 @@ class ChromeEventListener extends goog.Disposable {
...
     };
-    this.chromeEvent_.addListener(this.callback_);
+    try {
+      this.chromeEvent_.addListener(this.callback_);
+    } catch (error) {
+      this.callback_ = null;
+      goog.log.warning(
+          logger, `Failed to subscribe to the event ${nameForLogs}: ${error}`);
+    }
   }
```
Comment (emaxx-google) [inline smart_card_connector_app/src/chrome-api-provider.js:418]: nit: The function is becoming lengthy and hard to read. Consider extracting the lambda (the one that's assigned to `this.callback_` above) to a helper function.

Comment (ViktoriiaKovalova) [reply smart_card_connector_app/src/chrome-api-provider.js:431]: done

File: smart_card_connector_app/src/chrome-api-provider.js (hunk 1 above)
Comment (ViktoriiaKovalova) [reply smart_card_connector_app/src/chrome-api-provider.js]: done

**Review by emaxx-google — APPROVED** [review]
Thank you!

**Comment by emaxx-google** [comment]
@ViktoriiaKovalova Do you want a new release with this fix?

**Pull request merged**